### PR TITLE
templatize probes

### PR DESCRIPTION
**Pull request by bilalbokharee**

### CODE REVIEW

**Comment by stakater-user** [comment]
@bilalbokharee Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/587870fb44e24b7562fd9fe8bfcf133b42a451eb/checks) has Failed!

**Comment by stakater-user** [comment]
@bilalbokharee Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/cb9f0af5c8afec216ae76562b82145f68aa88c75/checks) has Failed!

**Comment by stakater-user** [comment]
@bilalbokharee Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/e43de8daa35e771f056508b2b87fd39310fc690b/checks) has Failed!

**Comment by stakater-user** [comment]
@bilalbokharee Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/5eab82d617d97e4fde5c72adbfb7addddb13f1e8/checks) has Failed!

**Comment by stakater-user** [comment]
@bilalbokharee Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater-charts/application/commit/e97ce3082663be9cbc1ba8c050476129f985b0a3/checks) has Failed!

**Comment by stakater-user** [comment]
@bilalbokharee Validation Successful

File: README.md [hunk 1]
```diff
@@ -102,13 +102,20 @@ To uninstall the chart:
 
 | Name                     | Description                                                                                  | Value           |
 | ------------------------ | -------------------------------------------------------------------------------------------- | --------------- |
-| deployment.probes.readinessProbe | The readiness probe                                                                  | See below       |
-| deployment.probes.livenessProbe| The livenessness probe                                                                 | See below       |
+| deployment.probes.readinessProbe | The readiness probe                                                                  |  See below      |
```
Comment (hanzala1234) [inline README.md:105]: remove this

File: README.md [hunk 2]
```diff
@@ -121,16 +128,29 @@ To uninstall the chart:
 
 ##### Liveness Probe
 
+| Name                     | Description                                                                                  | Value           |
+| ------------------------ | -------------------------------------------------------------------------------------------- | --------------- |
+| deployment.probes.livenessProbe| The livenessness probe                                                                 | See below       |
```
Comment (hanzala1234) [inline README.md:133]: remove this

File: README.md [hunk 3]
```diff
@@ -121,16 +128,29 @@ To uninstall the chart:
...
+| deployment.probes.livenessProbe.periodSeconds | Perform probe  everytime after specified periodSeconds                                                                  | 10       |
+| deployment.probes.livenessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed.                                                                  | 1       |
+| deployment.probes.livenessProbe.timeoutSeconds | Number of seconds after which the probe times out.                                                                  | 1       |
+| deployment.probes.livenessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness or readiness probes are initiated.                                                                  | 10       |
+| deployment.probes.livenessProbe.exec | Kubelet executes the command to perform the probe                                                                  | See below       |
```
Comment (hanzala1234) [inline README.md:140]: default value should be {}

File: README.md [hunk 4]
```diff
@@ -102,13 +102,20 @@ To uninstall the chart:
...
+| deployment.probes.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out.                                                                  | 1       |
+| deployment.probes.readinessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness or readiness probes are initiated.                                                                  | 10       |
+| deployment.probes.readinessProbe.httpGet | Describes an action based on HTTP Get requests                                                                  | See below       |
 
 ##### Readiness Probe
 
 ```
     readinessProbe:
+      enabled: true
       failureThreshold: 3
```
Comment (hanzala1234) [inline README.md:119]: we should only have httpGet example here. since see below is just in httpGet

File: README.md [hunk 5]
```diff
@@ -102,13 +102,20 @@ To uninstall the chart:
...
+| deployment.probes.readinessProbe.periodSeconds | Perform probe  everytime after specified periodSeconds                                                                  | 10       |
+| deployment.probes.readinessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed.                                                                  | 1       |
+| deployment.probes.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out.                                                                  | 1       |
+| deployment.probes.readinessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness or readiness probes are initiated.                                                                  | 10       |
+| deployment.probes.readinessProbe.httpGet | Describes an action based on HTTP Get requests                                                                  | See below       |
```
Comment (hanzala1234) [inline README.md:113]: exec is not defined

File: README.md [hunk 6]
```diff
@@ -121,16 +128,29 @@ To uninstall the chart:
...
+      enabled: true
       failureThreshold: 3
       periodSeconds: 10
       successThreshold: 1
       timeoutSeconds: 1
       initialDelaySeconds: 10
-      httpGet:
-        path: /path
-        port: 8080
+      exec:
```
Comment (hanzala1234) [inline README.md:150]: default exec is {}.right?

File: README.md [hunk 7]
```diff
@@ -117,12 +124,25 @@ To uninstall the chart:
...
         path: /path
         port: 8080
+      exec: {}
 ```
 
 ##### Liveness Probe
 
+| Name                     | Description                                                                                  | Value           |
+| ------------------------ | -------------------------------------------------------------------------------------------- | --------------- |
+| deployment.probes.livenessProbe.enabled | Enabled readiness probe                                                                  | true       |
```
Comment (hanzala1234) [inline README.md:134]: liveness*

File: README.md [hunk 8]
```diff
@@ -131,7 +151,18 @@ To uninstall the chart:
       httpGet:
         path: /path
         port: 8080
+      exec: {}
 ```
+### Configuring probes
+
+In order to use probes in deployment, set enable to `true`. After configuring `failureThreshold`, `periodSeconds`, `successThreshold`, `timeoutSeconds`, `initialDelaySeconds`, configure handler type to either `httpGet` or `exec`. You can only use one handler in one probe. `httpGet` is enabled by default, other possible handler is `exec`, whose configuration is specified as below:
```
Comment (hanzala1234) [inline README.md:158]: After configuring `failureThreshold`, `periodSeconds`, `successThreshold`, `timeoutSeconds`, `initialDelaySeconds`

I think these are by default defined. so no need to mention it here

File: README.md [hunk 9]
```diff
@@ -102,13 +102,20 @@ To uninstall the chart:
...
+| deployment.probes.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out.                                                                  | 1       |
+| deployment.probes.readinessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness or readiness probes are initiated.                                                                  | 10       |
+| deployment.probes.readinessProbe.httpGet | Describes an action based on HTTP Get requests                                                                  |   path: '/path' port: 8080     |
+| deployment.probes.readinessProbe.exec | Kubelet executes the specified command to perform the probe                                                                  |   {}   |
 
 ##### Readiness Probe
 
 ```
     readinessProbe:
+      enabled: true
```
Comment (hanzala1234) [inline README.md:118]: I think this can be removed from here. and add it in section below

Comment (bilalbokharee) [reply README.md:118]: Its part of configuration so it should be same everywhere. Causes confusion

Comment (hanzala1234) [reply README.md:118]: I mean readinessProbe complete section should be removed

Comment (hanzala1234) [reply README.md:118]: whole section of it

**Review by rasheedamir — CHANGES_REQUESTED** [review]
update changelog part in the readme

**Review by hanzala1234 — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -511,11 +504,44 @@ In order to use environment variable in deployment or cronjob, you will have to
...
+```
+
+
  
 # Changelog
 
 All notable changes to this project will be documented here
 
+## v1.1.9
+- Feature: add functionality to disable liveness and readiness probes.
```
Comment (rasheedamir) [inline README.md:544]: isn't it more than that? we can also set individual values on the probes?

**Review by callum-stakater — APPROVED** [review]
lgtm

File: README.md [hunk 11]
```diff
@@ -511,11 +504,44 @@ In order to use environment variable in deployment or cronjob, you will have to
...
 
    **Note:** first key after ``envFrom`` is just used to uniquely identify different objects in ``envFrom`` block. Make sure to keep it unique and relevant 
+
+
+### Configuring probes
+
+By default probe handler type is `httpGet`. You just need to override `port` and `path` as per your need.
+
+```
+  livenessProbe:
```
Comment (rasheedamir) [inline README.md:514]: @bilalbokharee don't i have to first set enabled true?

File: README.md (hunk 10 above)
Comment (rasheedamir) [reply README.md:544]: previously we could set on full block and not individual values of the probes; plz double check

File: README.md (hunk 11 above)
Comment (bilalbokharee) [reply README.md:514]: This is an example of httpGet config only. Should I add enabled keyword here as well? This snippet does not indicate enabling.

Comment (rasheedamir) [reply README.md:514]: shouldn't examples be set based on our chart? if I am reading this example I will miss it easily and just copy and paste and then struggle

File: README.md [hunk 12]
```diff
@@ -511,11 +504,45 @@ In order to use environment variable in deployment or cronjob, you will have to
...
+    httpGet:
+      path: '/path'
+      port: 8080
+```
+
+
+In order to use `exec` handler, you can define field `livenessProbe.exec` in your values.yaml.
+
+```
+  livenessProbe:
```
Comment (rasheedamir) [inline README.md:524]: @bilalbokharee again here that should be set

File: README.md [hunk 13]
```diff
@@ -511,11 +504,45 @@ In order to use environment variable in deployment or cronjob, you will have to
...
+
+
  
 # Changelog
 
 All notable changes to this project will be documented here
 
+## v1.1.9
+- Feature: add functionality to disable liveness and readiness probes.
+- Feature: support `exec` handler type in liveness and readiness probes
```
Comment (rasheedamir) [inline README.md:545]: wasn't this available before as well?

**Review by rasheedamir — CHANGES_REQUESTED** [review]
look at the comments

File: README.md (hunk 13 above)
Comment (bilalbokharee) [reply README.md:545]: no

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**